Frends.HTTP.UploadFile/Frends.HTTP.UploadFile/Frends.HTTP.UploadFile.cs

```diff
@@ -0,0 +1,191 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Frends.HTTP.UploadFile
+{
+    /// <summary>
+    /// Represents a task that posts a file to a web API endpoint.
+    /// </summary>
+    public static class UploadFileTask
+    {
+        /// <summary>
+        /// Send file using StreamContent
+        /// </summary>
+        /// <param name="input">Input parameters</param>
+        /// <param name="options">Optional parameters with default values</param>
+        /// <param name="cancellationToken">The cancellation token that can be used to cancel the upload operation.</param>
+        /// <returns>Object with the following properties: JToken Body. Dictionary(string,string) Headers. int StatusCode</returns>
+        /// public static bool Delete([PropertyTab] string fileName, [PropertyTab] OptionsClass options)
+        public static async Task<object> UploadFile([PropertyTab] Input input, [PropertyTab] Options options, CancellationToken cancellationToken)
+        {
+            using (var handler = new HttpClientHandler())
+            {
+                handler.SetHandleSettingsBasedOnOptions(options);
+
+                using (var httpClient = new HttpClient(handler))
+                {
+                    var responseMessage = await GetHttpRequestResponseAsync(httpClient, input, options, cancellationToken).ConfigureAwait(false);
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    string body = string.Empty;
+                    IEnumerable<KeyValuePair<string, IEnumerable<string>>> contentHeaders = new Dictionary<string, IEnumerable<string>>();
+
+                    if (responseMessage.Content != null)
+                    {
+                        body = await responseMessage.Content.ReadAsStringAsync().ConfigureAwait(false);
+                        contentHeaders = responseMessage.Content.Headers;
+                    }
+                    var response = new Response
+                    {
+                        Body = body,
+                        StatusCode = (int)responseMessage.StatusCode,
+                        Headers = GetResponseHeaderDictionary((IEnumerable<KeyValuePair<string, IEnumerable<string>>>)responseMessage.Headers ?? new Dictionary<string, IEnumerable<string>>(), contentHeaders)
```

Comment:
Please get this line to be easier to read - separate it into more lines of code, but make it easier to understand what is happening.